fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -312,7 +312,8 @@ public GroupId getGroup(long tableId) {
     public Set<GroupId> getAllGroupIds() {
         readLock();
         try {
-            return group2Tables.keySet();
+            // make a copy set to avoid ConcurrentModificationException
+            return new HashSet<>(group2Tables.keySet());
```

Comment:
```suggestion
            return Sets.newHashSet(group2Tables.keySet());
```

Reply:
Why we must use the google api? Sets.newHashSet has an additional layer of judgment inside.

Reply:
It's not necessary.